### PR TITLE
build: minimize docker mage using multistage multi-stage builds

### PR DESCRIPTION
**Pull request by derekbit**
Reduce image size from 444 MiB to 268 MiB

Longhorn/longhorn#6022

### CODE REVIEW

File: package/Dockerfile [hunk 1]
```diff
@@ -36,35 +25,82 @@ RUN git clone https://github.com/longhorn/go-spdk-helper.git ${GO_SPDK_HELPER_DI
...
+    zypper --gpg-auto-import-keys ref
+
+RUN zypper -n install kmod curl fuse wget \
+    nfs-client nfs4-acl-tools cifs-utils \
+    libibverbs librdmacm1 rdma-core-devel perl-Config-General libaio-devel sg3_utils \
+    iputils telnet iproute2 qemu-tools \
+    e2fsprogs e2fsprogs-devel xfsprogs xfsprogs-devel util-linux-systemd python3
+
+# Install SPDK dependencies
+COPY --from=builder /usr/src/spdk/scripts /tmp/spdk
```
Comment (innobead) [inline package/Dockerfile:77]: So `/tmp/spdk/pkgdep.sh` depends on `/tmp/spdk`? What's content in scripts folder?

**Review by innobead — COMMENTED** [review]
In general, LGTM. Just one question.

File: package/Dockerfile (hunk 1 above)
Comment (derekbit) [reply package/Dockerfile:77]: `/tmp/spdk` contains the scripts and utilities for 
- dependencies installation
- rpc python files
- performance measurement
- ...

**Review by innobead — APPROVED** [review]

**Comment by innobead** [comment]
#@mergify backport v1.5.x

**Pull request merged**